db/commitlog/commitlog.cc

```diff
@@ -973,11 +986,22 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
             overhead += descriptor_header_size;
         }
 
-        auto a = align_up(s + overhead, _alignment);
+        s += overhead;
+        // add bookkeep data reqs. 
+        s += (s / _alignment) * detail::sector_overhead_size;
```

Comment:
_alignment - sector_overhead?

Reply:
We can cross several blocks, so need to add overhead for all.

Reply:
Don't understand. Each block only store _alignment - sector_overhead_size, so the number of blocks isn't s/_alignment.

Reply:
Oops. Another typo.

Reply:
Was a var, got lost somewhere.

Reply:
Well the tests didn't catch it, meaning they're nowhere near close to covering the computation.

Let's add a test that writes randomly sized messages and reads (and verifies) them again, and see that it fails with this bug.

Reply:
No need to random for that. The calculation will be fine until about data size >= 50x sector, iff sector == 512. And even then, the default/min buffer size of 128k makes it not fail. But a very large data check should catch it. Should maybe add an extra assert here as well.

Reply:
Well let's make sure there are tests that exercise the math.